### PR TITLE
[specific ci=Group1-Docker-Commands] fix false container name conflicts

### LINKED ISSUE #4020: Container name and ID are conflated for container naming
**VIC version:**

master

**Steps to reproduce:**

```
$ sudo docker run -itd busybox
6fa54b2d087a2322ad8aaafd9c541636891a6f6c7d737c7ca059d5f024d62030
$ sudo docker ps -a
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
6fa54b2d087a        busybox             "sh"                2 seconds ago       Up 2 seconds                            cocky_goldstine
$ sudo docker run -itd --name=6fa busybox
0a21a9ee8306bb400ec457c4ac1ffd3e6f0bb4fa83a1d467fdea9a539d95bd64
```

**Actual behavior:**

```
$ sudo docker create --name=6fa busybox
Error response from daemon: Conflict. The name "6fa" is already in use by container 6fa54b2d087a2322ad8aaafd9c541636891a6f6c7d737c7ca059d5f024d62030. You have to remove (or rename) that container to be able to re use that name.
```

**Expected behavior:**

```
$ sudo docker ps -a
CONTAINER ID        IMAGE               COMMAND             CREATED              STATUS              PORTS               NAMES
0a21a9ee8306        busybox             "sh"                37 seconds ago       Up 36 seconds                           6fa
6fa54b2d087a        busybox             "sh"                About a minute ago   Up About a minute                       cocky_goldstine
```

**Additional details as necessary:**

This issue was found due to a container name collision in an integration test, where a container's name prefix-matched with a previous container's ID. This is a valid scenario, however, and the second container should be created successfully.

See below for previous issue description.

---

_We should review names of containers in our robot files and make them more specific to avoid failures due to name collisions._

_Seen in https://ci.vmware.run/vmware/vic/8130:_

collision

### PR DESCRIPTION
**Pull request by kreamyx**
Fixes #4020

### CODE REVIEW

File: tests/resources/Docker-Util.robot [hunk 1]
```diff
@@ -124,6 +124,22 @@ Run Regression Tests
...
     Should Be Equal As Integers  ${rc}  0
 
+    # regression test for #4020
+    ${rc}  ${container1}=  Run And Return Rc And Output  docker %{VCH-PARAMS} run -itd busybox
+    Should Be Equal As Integers  ${rc}  0
+    ${shortID1}=  Get container shortID  ${container1}
+    ${rc}  ${container2}=  Run And Return Rc And Output  docker %{VCH-PARAMS} run -itd --name ${shortID1} busybox
+    Should Be Equal As Integers  ${rc}  0
+    Should Not Contain  ${container2}  Conflict
+    ${rc}=  Run And Return Rc  docker %{VCH-PARAMS} stop ${container1}
```
Comment (anchal-agrawal) [inline tests/resources/Docker-Util.robot]: We can replace the entire stop+rm block with

```
${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} rm -f ${container1} ${container2}
Should Be Equal As Integers  ${rc}  0
Should Not Contain  ${output}  Error
```

File: lib/apiservers/engine/backends/cache/container_cache.go [hunk 2]
```diff
@@ -61,13 +73,17 @@ func (cc *CCache) getContainer(nameOrID string) *container.VicContainer {
...
 		return container
 	}
-
-	if container, exist := cc.containersByName[nameOrID]; exist {
-		return container
-	}
 	return nil
 }
 
+// GetContainerByName function returns a container whose "exactly" matches nameOnly
```
Comment (anchal-agrawal) [inline lib/apiservers/engine/backends/cache/container_cache.go]: Perhaps:

// GetContainerByName function returns a container whose **name** "exactly" matches nameOnly

**Comment by anchal-agrawal** [comment]
Please consider changing your PR title / commit message (sans the `specific ci` keyword) - something like "Fix false container name conflicts" conveys the intent better.

**Comment by kreamyx** [comment]
@mhagen-vmware Done!

**Comment by anchal-agrawal** [comment]
Pinging @jzt for review since this involves changes in the persona's container cache.

**Comment by kreamyx** [comment]
@mhagen-vmware Could you please take a look at the modified tests and unblock the merge if you approve?

**Pull request merged**